octavia_proxy/api/drivers/elbv3/driver.py

```diff
@@ -559,19 +589,31 @@ def flavors(self, session, project_id, query_filter=None):
         query_filter.pop('project_id', None)
 
         result = []
-
-        for fl in session.vlb.flavors(**query_filter):
-            fl_data = _flavors.FlavorResponse.from_sdk_object(fl)
-            fl_data.provider = PROVIDER
-            result.append(fl_data)
-
+        if 'name' in query_filter:
+            query_filter['name'] = ''.join(
+                ('L7_flavor.elb.', query_filter['name'])
+            )
+        if 'id' in query_filter:
+            fl_data = self.flavor_get(
+                project_id=project_id, session=session,
+                fl_id=query_filter['id']
+            )
+            if fl_data:
+                result.append(fl_data)
+        else:
+            for fl in session.vlb.flavors(**query_filter):
+                if not fl['name'].startswith('L4'):
+                    fl['name'] = re.sub(r'L\d_\w+.\w+.', '', fl['name'])
+                    fl_data = _flavors.FlavorResponse.from_sdk_object(fl)
+                    fl_data.provider = PROVIDER
+                    result.append(fl_data)
         return result
 
-    def flavor_get(self, session, fl_id):
+    def flavor_get(self, session, project_id, fl_id):
         LOG.debug('Searching flavor')
-
         fl = session.vlb.find_flavor(
             name_or_id=fl_id, ignore_missing=True)
+        fl['name'] = re.sub(r'L\d_\w+.\w+.', '', fl['name'])
```

Comment:
don't know how this ended up here, but XXX_get is not for search by name, but only fetch using ID. This means no find_flavor and no regexps